engine/battlehack20/stubs.py

```diff
@@ -0,0 +1,121 @@
+from typing import List, Optional, Tuple, Union
+
+from battlehack20.engine.game.game import Team, RobotType
+
+# The stubs in this file make it possible for editors to auto-complete the global methods
+# They can be imported using "from battlehack20.stubs import *"
+# This import is preprocessed away before instrumenting the code
+
+
+def log(msg: str) -> None:
+    """
+    Type-agnostic method.
+
+    Logs a message.
+    """
+    log(msg)
```

Comment:
Is there a reason you're recursing here instead of e.g. `pass`?

Reply:
In the case of `log` the reason is consistency with the other methods. When I used `pass` on the methods which have a return type other than `None`, VS Code started annoying me with a warning like "this method is supposed to return something but there is no return inside it".

Reply:
ohh I see. lol then I like this hack. could you please add a comment explaining this at the top of the file?

Reply:
Done :)